eclipse-theia/theia
# deps: upgrade+update

#### What it does

Upgrade most of our dependencies, and bump some of them in order to bring `yarn audit` from 16k issues down to 0.

This is roughly equivalent to deleting `yarn.lock` and generating a fresh one. To whomever concerned by this: Clients were never dependent on our lock file: they always pull the most recent versions based on their own version ranges and their own lock files.

Most conflicts were typing issues, and some vulnerabilities required me to bump some dependencies.

The good thing is that thanks to this, we'll keep developing with up-to-date dependencies, and we can affirm that according to `yarn audit`, the framework isn't distributed with widely-spread vulnerabilities.

Fixes https://github.com/theia-ide/theia/issues/5952

#### How to test

`yarn audit` should report 0 issues.
`yarn build` and `yarn test` should both complete successfully.
`@theia/example-*` applications should still run.

#### Review checklist


#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
